nmproxy/nm-proxy.go

```diff
@@ -46,14 +46,21 @@ func Start(ctx context.Context, wg *sync.WaitGroup,
 	if err != nil {
 		logger.FatalLog("failed to create proxy: ", err.Error())
 	}
+	proxyWaitG := &sync.WaitGroup{}
```

Comment:
add a comment as to why the extra waitgroup is required.

Reply:
we have cleanups happening in the `nmproxy.Start()` on shutdown, which should  only happen after all the threads in the proxy are closed and to do this parent wait group cannot be used to wait